design-documents/service-isolation/detailed-architecture.md

```diff
@@ -0,0 +1,87 @@
+# Detailed architecture of multi-service Magento
+
+## Overview
+
+The goal of this document is to gather requirements, understand data flow and propose architecture for multi-service Magento application.
+
+Please make sure to get familiar with [high-level service isolation vision](../service-isolation.md) before proceeding.
+
+![Gateway](img/GraphQL%20BFF.png)
+
+The client (e.g. browser) is interacting with the system using GraphQL. The only component of the system exposed publicly is Storefront Gateway, which is responsible for routing, full response caching and perimeter-level authorization.
+
+Storefront Gateway routes incoming queries between PWA, GraphQL Engine and Storefront Authentication Gateway.
```

Comment:
For all internal APIs whether storefront or admin, should we need to distinguish between the two? or just say API Gateway? which could later on be the gateway for Admin APIs as well,

Reply:
I believe Admin API also will be exposed because it's needed to build integration by 3rd party applications to work with Store Management application.

Reply:
It is possible to specify custom Admin base URL. In this case we will have to have two gateways, right?
If not needed on specific installation, we can combine them for simplicity.

Reply:
yes I would suggest to combine, unless there is a solid benefit and sometimes security is that concern; but if none at the moment, then why maintaining another infrastructure layer.

Reply:
Mentioned this in the proposal.